sql/src/main/java/org/apache/druid/sql/guice/BrokerServiceModule.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.guice;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Provides;
+import org.apache.druid.discovery.DruidNodeDiscoveryProvider;
+import org.apache.druid.discovery.NodeRole;
+import org.apache.druid.guice.LazySingleton;
+import org.apache.druid.guice.ManageLifecycle;
+import org.apache.druid.guice.annotations.EscalatedGlobal;
+import org.apache.druid.guice.annotations.Json;
+import org.apache.druid.rpc.DiscoveryServiceLocator;
+import org.apache.druid.rpc.ServiceClientFactory;
+import org.apache.druid.rpc.ServiceLocator;
+import org.apache.druid.rpc.StandardRetryPolicy;
+import org.apache.druid.rpc.guice.ServiceClientModule;
+import org.apache.druid.sql.client.Broker;
+import org.apache.druid.sql.client.BrokerClient;
+import org.apache.druid.sql.client.BrokerClientImpl;
+
+/**
+ * Module that processes can bind to if they require a {@link BrokerClient}.
+ * <p>
+ * This extend {@link ServiceClientModule} because the {@link BrokerClient} requires
+ * classes present in the sql module.
+ * </p>
+ */
+public class BrokerServiceModule extends ServiceClientModule
```

Comment:
Nit:
Why extend `ServiceClientModule`?
I think the only required method from `ServiceClientModule` is `makeServiceClientFactory()`.
I guess it is okay to copy over that method.

Reply:
yeah, `makeServiceClientFactory()` and the constants for number of retries and connection threads to use. So I left it to avoid duplication.

Reply:
Instead of extending, you could add a static method in `ServiceClientModule` which is used both by `ServiceClientModule` and this new `BrokerServiceModule`.

The reason I am unsure about extending is it is a little confusing.
- Firstly we don't need the overlord/coordinator methods exposed by `ServiceClientModule`.
- Secondly, if we install both modules, which one ends up providing the `OverlordClient`/`CoordinatorClient` implementation.

Reply:
That's a good point. Added a static method per your suggestion, thanks!